src/XMLimport.cpp

```diff
@@ -172,12 +172,18 @@ bool XMLimport::importPackage(QFile* pfile, QString packName, int moduleFlag, QS
                     /*||(mVersionMajor==1&&mVersionMinor)*/) {
                     // Minor check is not currently relevant, just abort on 2.000f or more
 
-                    QString moanMsg = tr("[ ALERT ] - Sorry, the file being read:\n"
-                                         "\"%1\"\n"
-                                         "reports it has a version (%2) it must have come from a later Mudlet version,\n"
-                                         "and this one cannot read it, you need a newer Mudlet!")
-                                              .arg(pfile->fileName(), versionString);
-                    mpHost->postMessage(moanMsg);
+                    if (pErrorMessage) {
+                        *pErrorMessage = QStringLiteral("failed to import package as it has a version number of %1 which is too late to "
+                                                        "be understood by this version of Mudlet, it seems you need a newer one")
+                                         .arg(versionString);
+                    } else {
+                        QString moanMsg = tr("[ ALERT ] - Sorry, the file being read:\n"
+                                             "\"%1\"\n"
+                                             "reports it has a version (%2) it must have come from a later Mudlet version,\n"
```

Comment:
Same here. What's an "early Mudlet" and a "late Mudlet"?

Reply:
"early" in this context would be an _old, antique_ version in a future time when it has been asked to read the files of a _less-old_ version.  The latter is a "later" version but given that the timeframe when this error message is displayed it is newer that the instance of Mudlet that is executing this code - but it need not be the "current" one. 😖

Reply:
Let's go with "older" and "newer", I think that would make more sense!

Reply:
The text is:

    "[ ALERT ] - Sorry, the file being read:\n"
    "\"%1\"\n"
    "reports it has a version (%2) it must have come from a later Mudlet version,\n"
    "and this one cannot read it, you need a newer Mudlet!"

___In this context___ it is reasonable to use _later_ to describe the position in time of the edition of Mudlet that created this file that cannot be read by the _currently executing_  edition.

Reply:
I'd rephrase the text as the following. Just a grammatical fix though.

```"[ ALERT ] - Sorry, the file being read:\n"
"\"%1\"\n"
"reports that it is version (%2). The file must have come from a later Mudlet version,\n"
"as this one cannot read it. You need a newer version of Mudlet to install this file!"```